src/App.tsx

```diff
@@ -1,9 +1,25 @@
-import React from 'react';
 import './App.scss';
+import { useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-export const App: React.FC = () => {
+export type Movie = (typeof moviesFromServer)[0];
+
+export const App = () => {
+  const [filteredMovieList, setFilteredMovieList] =
+    useState<Movie[]>(moviesFromServer);
+
+  function handleInputTextChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const searchQuery = event.target.value.toLowerCase().trim();
+    const filteredMovies = moviesFromServer.filter(
+      movie =>
+        movie.title.toLowerCase().includes(searchQuery) ||
+        movie.description.toLowerCase().includes(searchQuery),
```

Comment:
Ensure that the filtering logic aligns with the task requirements. If the task specifies certain fields to be included in the search, verify that these fields are correctly used in the filter function.